docs/learn/prompts.md

```diff
@@ -31,9 +53,13 @@ print(prompt)
 # > Recommend a fantasy book
 ```
 
-For your convenience, we will also automatically dedent and strip the prompt template. This makes writing multi-line prompts a breeze:
+In this example, the `@prompt_template` decorator defines the template string `"Recommend a {genre} book"`. The genre attribute of the `BookRecommendationPrompt` class corresponds to the `{genre}` placeholder in the template. When you create an instance of the class with `genre="fantasy"`, this value is automatically injected into the template.
```

Comment:
hmm, this feels entirely duplicated from the previous section? we should merge this into the explanation above instead of duplicating the code?